app/controllers/publish/providers/study_sites_controller.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Publish
+  module Providers
+    class StudySitesController < PublishController
```

Comment:
Can this controller also re-use same logic as the `SchoolsController`?

Reply:
I am not sure that is the way to go, we could do a module like so: 
```
module Locations
  extend ActiveSupport::Concern

  included do
    before_action :pundit
    before_action :site, only: %i[show delete]
    before_action :build_site, only: %i[new create]
  end

  def show; end

  def new
    @school_form = ::Support::SchoolForm.new(provider, @site, params: gias_school_params)
    @school_form.clear_stash
  end

  def edit
    @school_form = ::Publish::SchoolForm.new(site)
  end

  def create
    @school_form = ::Support::SchoolForm.new(provider, @site, params: site_params(:support_school_form))
    if @school_form.stash
      redirect_to publish_provider_recruitment_cycle_check_school_path
    else
      render :new
    end
  end

  def update
    @school_form = ::Publish::SchoolForm.new(site, params: site_params(:publish_school_form))

    if @school_form.save!
      course_updated_message('School')

      redirect_to publish_provider_recruitment_cycle_school_path(
        @school_form.provider_code, @school_form.recruitment_cycle_year, site.id
      )
    else
      render :edit
    end
  end

  def delete; end

  def destroy
    site.destroy!
    flash[:success] = 'School removed'
    redirect_to publish_provider_recruitment_cycle_schools_path
  end

  private

  def pundit
    authorize provider, :show?
  end

  def site_params(param_form_key)
    params.require(param_form_key).permit(::Publish::SchoolForm::FIELDS)
  end

  def gias_school_params
    return {} unless params[:school_id] || params[:study_site_id]

    gias_school.school_attributes
  end

  def gias_school
    @gias_school ||= GiasSchool.find(params[:school_id])
  end
end
```
but we still would need conditional redirects on create, update and destroy. And the form models would share the same name. To my eye cognitive load in following this is more than having seperate controllers?

Reply:
Yup fair enough